--- a/src/drivers/driver_nl80211.c
+++ a/src/drivers/driver_nl80211.c
@@ -1674,8 +1674,15 @@
 		os_free(rcfg);
 	}
 
-	if (linux_iface_up(drv->global->ioctl_sock, ifname) > 0)
+	int result;
+
+	result = linux_iface_up(drv->global->ioctl_sock, ifname);
+	if (result == -ENODEV) {
+		goto failed;
+	}
+	if (result > 0) {
 		drv->start_iface_up = 1;
+	}
 
 	if (wpa_driver_nl80211_finish_drv_init(drv, set_addr, 1, driver_params))
 		goto failed;
